server/src/main/webapp/token/index.html

```diff
@@ -0,0 +1,66 @@
+<!--
+  ~  Copyright (c) 2009 - ${YEAR} Red Hat, Inc.
+  ~  
+  ~  This software is licensed to you under the GNU General Public License,
+  ~  version 2 (GPLv2). There is NO WARRANTY for this software, express or
+  ~  implied, including the implied warranties of MERCHANTABILITY or FITNESS
+  ~  FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+  ~  along with this software; if not, see
+  ~  http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+  ~  
+  ~  Red Hat trademarks are not licensed under GPLv2. No permission is
+  ~  granted to use or replicate Red Hat trademarks that are incorporated
+  ~  in this software or its documentation.
+  -->
+
+<!DOCTYPE html>
+<html lang="en">
+<head>
+        <meta charset="utf-8"/>
+</head>
+<body>
+<script src="keycloak.js"></script>
+<script>
+        var status;
+        var xhttp = new XMLHttpRequest();
+        var host = window.location.hostname;
+        xhttp.open('GET', "https://" + host + ":8443/candlepin/status", true);
+        xhttp.onreadystatechange = function() {
+                var realm;
+                var authUrl;
+                var clientId;
+
+                if (xhttp.readyState == 4) {
+                        if (xhttp.status == 200) {
+                                var status = JSON.parse(xhttp.responseText);
+                                realm = status['realm']
+                                authUrl = status['authUrl']
+                                clientId = status['resource']
+                        }
+                        var keycloak = Keycloak({
+                                url: authUrl,
+                                realm: realm,
+                                clientId: clientId
+                        });
```

Comment:
When xhttps.status is not 200, then `clientId` is undefined. BTW: Canldepin returned following result for `/candlepin/status`:

```json
{
  "mode" : "NORMAL",
  "modeReason" : "STARTUP",
  "modeChangeTime" : "2019-08-07T12:02:54+0000",
  "result" : true,
  "version" : "2.8.0",
  "rulesVersion" : "5.37",
  "release" : "1",
  "standalone" : true,
  "timeUTC" : "2019-08-07T12:21:54+0000",
  "realm" : null,
  "authUrl" : null,
  "resource" : null,
  "rulesSource" : "default",
  "managerCapabilities" : [ "instance_multiplier", "derived_product", "vcpu", "cert_v3", "hypervisors_heartbeat", "remove_by_pool_id", "syspurpose", "storage_band", "cores", "hypervisors_async", "org_level_content_access", "guest_limit", "ram", "batch_bind" ]
}
```

Reply:
Did you enable keycloak auth in candlepin.conf?